shell/platform/windows/flutter_windows_engine_unittests.cc

```diff
@@ -95,6 +96,63 @@ TEST(FlutterWindowsEngine, RunDoesExpectedInitialization) {
   modifier.embedder_api().Shutdown = [](auto engine) { return kSuccess; };
 }
 
+TEST(FlutterWindowsEngine, RunWithoutANGLEUsesSoftware) {
+  std::unique_ptr<FlutterWindowsEngine> engine = GetTestEngine();
+  EngineEmbedderApiModifier modifier(engine.get());
+
+  // The engine should be run with expected configuration values.
+  bool run_called = false;
+  modifier.embedder_api().Run = MOCK_ENGINE_PROC(
+      Run, ([&run_called, engine_instance = engine.get()](
+                size_t version, const FlutterRendererConfig* config,
+                const FlutterProjectArgs* args, void* user_data,
+                FLUTTER_API_SYMBOL(FlutterEngine) * engine_out) {
+        run_called = true;
+        *engine_out = reinterpret_cast<FLUTTER_API_SYMBOL(FlutterEngine)>(1);
+        // We don't have an AngleSurfaceManager, so we should be using software.
+        EXPECT_EQ(config->type, kSoftware);
+        return kSuccess;
+      }));
+
+  // It should send locale info.
+  bool update_locales_called = false;
+  modifier.embedder_api().UpdateLocales = MOCK_ENGINE_PROC(
+      UpdateLocales,
+      ([&update_locales_called](auto engine, const FlutterLocale** locales,
+                                size_t locales_count) {
+        update_locales_called = true;
+
+        EXPECT_GT(locales_count, 0);
+        EXPECT_NE(locales, nullptr);
+
+        return kSuccess;
+      }));
+
+  // And it should send initial settings info.
+  bool settings_message_sent = false;
+  modifier.embedder_api().SendPlatformMessage = MOCK_ENGINE_PROC(
+      SendPlatformMessage,
+      ([&settings_message_sent](auto engine, auto message) {
+        if (std::string(message->channel) == std::string("flutter/settings")) {
+          settings_message_sent = true;
+        }
+
+        return kSuccess;
+      }));
```

Comment:
Consider checking the `enable_software_rendering` setting in the message here.

Reply:
So it looks like neither the Embedder API nor the Windows embedder use this setting.